server/bin/pbench-backup-tarballs.py

```diff
@@ -49,16 +49,16 @@ def __init__(
 
 def sanity_check(lb_obj, s3_obj, config, logger):
     # make sure archive is present
-    archive = config.ARCHIVE
-    archivepath = os.path.realpath(archive)
+    archive = Path(config.ARCHIVE)
+    archivepath = archive.resolve()
 
     if not archivepath:
```

Comment:
I believe this is a pre-existing bug, as `realpath` does not return an empty string if the given path does not exist.

With `pathlib`, we have an opportunity to fix that, with something like:

```
archive = Path(config.ARCHIVE)
try:
    archivepath = archive.resolve(strict=True)
except FileNotFound:
    logger.error("The ARCHIVE directory, '{}', does not resolve to a real location", archive)
else:
    if not archivepath.is_dir():
        logger.error("The ARCHIVE directory, '{}', does not resolve to a directory ('{}')", archive, archivepath)
```

Reply:
Ok sure